internal/access/credential.go

```diff
@@ -225,3 +234,43 @@ func checkPasswordRequirements(db data.ReadTxn, password string) error {
 
 	return nil
 }
+
+// checkPasswordBreached checks if the password is found in previous data breaches. It uses
+// Have I Been Pwned's API to perform this check. The API takes the first 5-characters prefix
+// of the SHA1 checksum of the password and returns a carriage returned list. A password is
+// breached if the remaining characters of its SHA1 checksum appears in this list.
+// Ref: https://haveibeenpwned.com/API/v3#PwnedPasswords
+func checkPasswordBreached(password string) error {
+	//nolint:gosec // this is required for the API
+	sum := sha1.Sum([]byte(password))
+	hexsum := hex.EncodeToString(sum[:])
+
+	request, err := http.NewRequestWithContext(context.Background(), http.MethodGet, fmt.Sprintf("https://api.pwnedpasswords.com/range/%s", hexsum[:5]), nil)
+	if err != nil {
+		return err
+	}
+
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return err
+	}
+
+	for _, line := range strings.Split(string(body), "\r\n") {
```

Comment:
I'd suggest https://pkg.go.dev/bufio#Scanner here instead of `ReadAll` and `Split`. 

That should use a lot less memory (because strings are read in small batches), and also avoids reading the rest of the response if something matches.

There are a few examples of `bufio.Scanner` in our repo already, I use it a bunch. It can accept the `response.Body` directly.